CoreScriptsRoot/Modules/Settings/Utility.lua

```diff
@@ -211,9 +217,7 @@ local function isTenFootInterface()
 end
 
 local function usesSelectedObject()
-	if UserInputService.TouchEnabled and not UserInputService.GamepadEnabled then return false end
-
-	return true
+	return not (UserInputService.TouchEnabled and not UserInputService.GamepadEnabled or UserInputService.VREnabled)
```

Comment:
I have to agree; this file seems to be full of difficult-to-parse logical statements. I'm in favor of splitting this up until multiple lines that spell out what's good and what isn't, even if it's ever so slightly more code.